src/shared/firewall-util-nft.c

```diff
@@ -0,0 +1,906 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+
+#include <arpa/inet.h>
+#include <endian.h>
+#include <errno.h>
+#include <stddef.h>
+#include <string.h>
+#include <linux/netfilter/nf_tables.h>
+#include <linux/netfilter/nf_nat.h>
+#include <linux/netfilter_ipv4.h>
+#include <netinet/ip.h>
+
+#include "sd-netlink.h"
+
+#include "alloc-util.h"
+#include "firewall-util.h"
+#include "firewall-util-private.h"
+#include "in-addr-util.h"
+#include "macro.h"
+#include "socket-util.h"
+#include "time-util.h"
+
+#define NFT_SYSTEMD_DNAT_MAP_NAME "map_port_ipport"
+#define NFT_SYSTEMD_TABLE_NAME   "io.systemd.nat"
+#define NFT_SYSTEMD_MASQ_SET_NAME "masq_saddr"
+
+#define NFNL_DEFAULT_TIMEOUT_USECS 1ULL * USEC_PER_SEC
+
+#define UDP_DPORT_OFFSET 2
+
+static int nfnl_netlink_sendv(sd_netlink *nfnl,
+                              sd_netlink_message *messages[],
+                              size_t msgcount) {
+        _cleanup_free_ uint32_t *serial = NULL;
+        unsigned int i;
```

Comment:
doesnt't matter, but we usually use just "unsigned" rather than "unsigned int"

Reply:
that said, this should actually be size_t I guess, given that you just count through `msgcount` elements and that param is rightfully size_t